vcpkg/users/binarycaching-troubleshooting.md

```diff
@@ -0,0 +1,164 @@
+---
+title: Binary caching troubleshooting guide
+description: Troubleshooting your binary caching setup
+author: dan-shaw
+ms.author: danshaw2
+ms.date: 9/20/2023
+ms.prod: vcpkg
+---
+
+# Binary caching troubleshooting guide
+
+This guide is intended for vcpkg binary caching users that are running into unexpected behavior from this experience. Please review the [binary caching](./binarycaching.md) reference documentation for a basic understanding of what this feature does and how it works.
+
+## Set up vcpkg to print debugging information
+
+Debug mode outputs information such as the compiler verison and ABI hashes, which will help us to diagnose any binary caching issues.
+
+* [Classic mode](./classic-mode.md): add `--debug` to your command invocation.
+* CMake toolchain: set `-DVCPKG_INSTALL_OPTIONS="--debug"` in your `cmake` command call or in your `CMakePresets.json` file.
+* MSBuild/Visual Studio: set the property `VcpkgAdditionalInstallOptions` to `--debug`
+
+## Troubleshooting binary caching configuration issues
+
+There are many choices for a binary caching provider including AWS, NuGet, Azure. While each provider will have its own error messages, we expect the type of error and the actual fix will be the same across all providers. We will walk through a specific example of using NuGet as a binary cache provider.
+
+### Binary package ABI
+
+The full Application Binary Interface (ABI) of the binary package is in the debug output. For example, for zlib:
```

Comment:
```suggestion
For each binary package artifact, vcpkg calculates an ABI hash (Application Binary Interface). The ABI hash is then used as a key to refer to that particular binary artifact. If any changes occur in the package, its ABI hash its changed too.

In debug mode, the output includes the full ABI hash of each package.
For example, for package `zlib`:
```

We should preamble with why vcpkg calculates an ABI hash.

Reply:
added a link instead